src/CorrelationId/ICorrelationContextFactory.cs

```diff
@@ -9,8 +9,9 @@ public interface ICorrelationContextFactory
         /// Creates a new <see cref="CorrelationContext"/> with the correlation ID set for the current request.
         /// </summary>
         /// <param name="correlationId">The correlation ID to set on the context.</param>
+        /// /// <param name="header">The header used to hold the correlation ID.</param>
         /// <returns>A new instance of a <see cref="CorrelationContext"/>.</returns>
-        CorrelationContext Create(string correlationId);
+        CorrelationContext Create(string correlationId, string header);
```

Comment:
BTW, this is a breaking change (why I'm here) and would've probably warranted a major bump 😝

Reply:
Crap! I actually didn't spot / consider this as generally I only expect the middleware to use the factory to create the context. I can see the need in unit tests though! Sorry about that! I'll add a note to the release notes. Can I just confirm how you're using the factory to make sure I have my bases covered?

Reply:
In our case, we did some mocking of `ICorrelationContextFactory`, it seems 😕

Reply:
Hmm okay. I can see why that might be needed, although I would be interested if you're able to show a snippet of the test? 

I'm going to publish a post about testing against this library. The approach I've taken is to create a fake accessor that I can pass into dependant objects. A bit like this:

```
public class FakeCorrelationIdAccessor : ICorrelationContextAccessor
{
	public CorrelationContext CorrelationContext
	{
		get => new CorrelationContextFactory().Create("FakeCorrelationId", "FakeHeader");
		set => throw new NotImplementedException();
	}
}
```

Reply:
Would a patch be useful that adds back the original single param Create method which will just set an empty value on the header if used?

Reply:
Ouch sorry for the breaking changes, I made the same assumption that only the middleware would use the factory.

Maybe the Create method with a single param should sets the default header name if used, it will be consistent with the default behavior.

Reply:
I don't think this test was useful anyway. I have no clue why it was written the way it was. In our case, we'll probably just change the tests and move on. I just wanted to raise this in case other were broken as well 😄

Reply:
Thanks for raising it @khellang. Definitely worth considering.

@juchom No worries! I made the same assumption and didn't really consider the impact in testing cases. I've started on a small patch that basically does what you suggest.